### PR TITLE
feat: Add script for dumping use-cases to disk for documentation

### PR DESCRIPTION
**Pull request by nalanj**
## Describe your changes

Adds a new script for dumping the data we need for use-cases in the repo to disk. A near-term PR will support running it alongside our flows.yaml sync.

## Checklist before requesting a review (skip if just adding/editing APIs & templates)

-   [ ] I added tests, otherwise the reason is:
-   [ ] External API requests have `retries`
-   [ ] Pagination is used where appropriate
-   [ ] The built in `nango.paginate` call is used instead of a `while (true)` loop
-   [ ] Third party requests are NOT parallelized (this can cause issues with rate limits)
-   [ ] If a sync requires metadata the `nango.yaml` has `auto_start: false`
-   [ ] If the sync is a `full` sync then `track_deletes: true` is set
-   [ ] I followed the best practices and guidelines from the [Writing Integration Scripts](/NangoHQ/integration-templates/blob/main/WRITING_INTEGRATION_SCRIPTS.md) doc

### CODE REVIEW

**Review by TBonnin — APPROVED** [review]
I am out of the loop regarding what we want to be doing. Code looks ok though

File: scripts/generate-docs-use-cases.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                endpoints.push({
+                    method: endpoint.method,
+                    path: endpoint.path,
+                    description: sync.description?.trim(),
+                    group: endpoint.group,
+                    script: `${integration}/syncs/${key}`
+                });
+            }
+        });
+    }
```
[the remark this answers is not in the document]
Comment (nalanj) [reply scripts/generate-docs-use-cases.ts]: See fffae023

File: scripts/generate-docs-use-cases.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                endpoints.push({
+                    method: endpoint.method,
+                    path: endpoint.path,
+                    description: action.description?.trim(),
+                    group: endpoint.group,
+                    script: `${integration}/actions/${key}`
+                });
+            }
+        });
+    }
```
[the remark this answers is not in the document]
Comment (nalanj) [reply scripts/generate-docs-use-cases.ts:42]: See fffae023

**Comment by nalanj** [comment]
@TBonnin just to bring you a little more in the loop, I'm dumping this file so that I can sync it over to the nango repo and use it to show all of the endpoints for the integrations within the new generated documentation.

**Comment by TBonnin** [comment]
> @TBonnin just to bring you a little more in the loop, I'm dumping this file so that I can sync it over to the nango repo and use it to show all of the endpoints for the integrations within the new generated documentation.

don't we already have those info in the `flows.yaml` file?

**Comment by nalanj** [comment]
@TBonnin I guess we do! I didn't realize that's what flows.yaml was.

**Pull request merged**